snuba/clickhouse/query.py

```diff
@@ -12,12 +13,16 @@ def __init__(self,
         # body anymore.
         request: Request,
         prewhere_conditions: Sequence[str],
-        final: bool,
+        query_hints: QueryHints,
     ) -> None:
         self.__dataset = dataset
         self.__request = request
         self.__prewhere_conditions = prewhere_conditions
-        self.__final = final
+        self.__query_hints = query_hints
+
+        if self.__query_hints.turbo:
+            if self.__request.query.get_sample() is None:
+                request.query.set_sample(settings.TURBO_SAMPLE_RATE)
```

Comment:
Can you process this condition when you actually read the sample field from query? So that you do not ahve to mutate the Snuba query when you are formatting the Clickhouse one ?
I would like to make this class more immutable.